### PR TITLE
Create Form shared UI component

### PR DESCRIPTION
**Pull request by uskayyyyy**
## Overview

This PR implements the `Form` component that can be used across the entire project. In this PR, the `Form` component is not applied to any existing components since the main goal of this PR is to create the component.

## Changes

- Created the `Form` component.
- Created the usage example of the `Form` component.
- Added the exclamation icon.

## Review points

- The example file does not accurately reflect the actual feature. So, please dismiss the minor problems in that file if any.

## Screen Captures

How it behaves:

https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/b639c6c2-88f5-4be7-b2b5-85de3682f494

An example of the submitted values (`console.log()` in the `proccessSubmit()` function):
Screenshot 2024-01-12 at 0 48 47

## References
- [[shadcn/ui] React Hook Form](https://ui.shadcn.com/docs/components/form)
- [[shadcn/ui] React Hook Form > Examples](https://ui.shadcn.com/docs/components/form#examples)
- [[YouTube] Using React Hook Form, Zod Validation and Server Actions in NextJs 13](https://youtu.be/R_Pj593TH_Q?si=kkIQTo1UGZqvBPmT)

## Note

- Server actions can be called inside the `procesForm()` handler instead of passing it to the `action` attribute of the `<form>`.
  -> [[YouTube] Using React Hook Form, Zod Validation and Server Actions in NextJs 13](https://youtu.be/R_Pj593TH_Q?si=kkIQTo1UGZqvBPmT) 

## Assignee Checklist:

- [x] The base branch is correct (See: [Types of Branches](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] Code is well-commented and easy to understand
- [ ] UI changes accurately reflect the provided design

### CODE REVIEW

File: src/components/ui/Form.tsx
Comment (kanta1207) [reply src/components/ui/Form.tsx:115]: Oh, I realized now that you removed `FormDescription` component from original file, so we don't need description id. My question NO.1 is solved now.
Please just share with me about No.2, the intention on commenting out on line 115

File: src/components/ui/Form.tsx
Comment (kanta1207) [inline src/components/ui/Form.tsx:134]: I just wondered, could you kindly share with me the reason why you didn't use `items-center` to align both icon and message, and instead used `items-start`?
With using `items-center` to both icon and message, i guess it'd be like this↓
```tsx
<div className="mt-1 flex items-center gap-1.5">
        <Icon icon={ExclamationIcon} size={3.5} color="danger" />
        <p
          ref={ref}
          id={formMessageId}
          className={cn('text-danger leading-5', className)}
          {...props}
        >
          {body}
        </p>
      </div>
```
↓Original implementation
Screenshot 2024-01-13 at 0 14 20

↓ With `items-center`
Screenshot 2024-01-13 at 0 16 00

**Review by kanta1207 — COMMENTED** [review]
Thank you for the implementation!
The `Form` component looks like implemented efficiently, thank you for removing unnecessary components from original file!

Although I'm not so sure about the comment I left to request a change, it'd be just appreciated if you can take a look at them and share your thoughts! 🙌

File: src/components/ui/Form.tsx
Comment (uskayyyyy) [reply src/components/ui/Form.tsx:134]: Good question! It's because when the message spans more than two lines, the alignment becomes misaligned. So, `items-start` is necessary. That is why the `line-height` of the message and the height of the icon area have the same height.

File: src/components/ui/Form.tsx
Comment (uskayyyyy) [reply src/components/ui/Form.tsx:115]: I forgot to address the comment. Please let me research about ARIA first. I will update it after that.

**Review by kotaaaa — COMMENTED** [review]
LGTM, so far. You can merge after you got approval from Kanta's review. Thanks!

File: src/components/ui/Form.tsx
Comment (kanta1207) [reply src/components/ui/Form.tsx:134]: Makes sense ! Thank you for detailed explanation!

File: src/components/ui/Form.tsx
Comment (kanta1207) [reply src/components/ui/Form.tsx:115]: Thank you for reserching! I'll just leave my opinion below, please correct it or just ignore it on your convenience!
IMO, since we removed `FormDescription` from original file, we will not have description for our `Form` component, thus we don't need `aria-describedby` anymore.

**Review by kanta1207 — APPROVED** [review]
Thank you for the detailed explanation about the intention on `items-start`! 
I believe no matter how your research on `ARIA` will go, your solution will be decent one and there's no need to keep this PR waiting for approval anymore, it's looking good!

**Review by kotaaaa — APPROVED** [review]

File: src/components/ui/Form.tsx
Comment (uskayyyyy) [reply src/components/ui/Form.tsx]: The conclusion was reached in reference to the following item: [`7e5b438` (#167)](https://github.com/genesis-tech-tribe/nishiki-frontend/pull/167/commits/7e5b438f7b7d0a87df995192160dc5a221e0bd51).

**Comment by uskayyyyy** [comment]
@kanta1207 @kotaaaa 
Thank you for your reviews!!

**Pull request merged**
